lib/btc/BtcRpc.js

```diff
@@ -53,65 +64,81 @@ class BtcRpc {
     if (passphrase === undefined) {
       passphrase = await passwordPromptAsync('> ');
     }
-    await this.walletUnlock({ passphrase, time: 10 });
+    await this.walletUnlock({ passphrase, time: 10800 });
     const tx = await this.sendToAddress({ address, amount });
     await this.walletLock();
     return tx;
   }
 
-  async unlockAndSendToAddressMany({ payToArray, passphrase, time = 10800 }) {
-    function RpcException(message) {
-      const e = new Error(message);
-      e.name = 'failedRequest';
-      e.data = errorObject;
-      return e;
-    }
-
+  async unlockAndSendToAddressMany({ payToArray, passphrase, time=10800, maxValue=10*1e8, maxOutputs=1 }) {
     if (passphrase === undefined) {
       passphrase = await passwordPromptAsync('> ');
     }
+    if (payToArray.length < maxOutputs) {
+      maxOutputs = payToArray.length;
+    }
     await this.walletUnlock({ passphrase, time });
-    this.emitter.emit('unlocked', time);
-    let someRequestFailed;
-    const errorObject = { success: {}, failure: {} };
-    const result = [];
-    for (const [i, request] of payToArray.entries()) {
-      const { address, amount } = request;
-      const emitData = {
-        address: request.address,
-        amount: request.amount,
-        id: request.id
+    let payToArrayResult = [];
+    while(payToArray.length) {
+      let currentValue = 0;
+      let currentOutputs = 0;
+      let batch = {};
+      let tempArray = [];
+      while(currentValue < maxValue && currentOutputs < maxOutputs) {
+        let tx = payToArray.shift();
+        tempArray.push(tx);
+        let address = tx.address;
+        let amount = tx.amount;
+        if (address in batch) {
```

Comment:
I would do something like 

```
if(batch[address] === undefined) {
  batch[address] = 0;
}
batch[address] += amount
```

The way you're doing it here will result in currentValue being increased more than it should for dupes.

IE [{"abc": 1}, {"abc": 1}] would result in currentValue being 1, then current value being 3